package/MDAnalysis/coordinates/array.py

```diff
@@ -0,0 +1,240 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- http://www.MDAnalysis.org
+# Copyright (c) 2006-2015 Naveen Michaud-Agrawal, Elizabeth J. Denning, Oliver Beckstein
+# and contributors (see AUTHORS for the full list)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""
+Reading trajectories from memory --- :mod:`MDAnalysis.coordinates.array`
+==========================================================================
+
+:Author: Wouter Boomsma
+:Year: 2016
+:Copyright: GNU Public License v2
+:Maintainer: Wouter Boomsma <wb@bio.ku.dk>, wouterboomsma on github
+
+
+.. versionadded:: 0.14.0
+
+The module contains a trajectory reader that operates on an array
+in memory, rather than reading from file. This makes it possible to
+use operate on raw coordinate using existing MDAnalysis tools. In
+addition, it allows the user to make changes to the coordinates in
+a trajectory (e.g. through AtomGroup.set_positions) without having
+to write the entire state to file.
+
+
+Examples
+--------
+
+Constructing a Reader from an array
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A simple example where a new universe is created from the
+array extracted from a DCD timeseries
+
+    from MDAnalysis import Universe
+    from MDAnalysisTests.datafiles import DCD, PDB_small
+    from MDAnalysis.coordinates.array import ArrayReader
+
+    universe = Universe(PDB_small, DCD)
+    coordinates = universe.trajectory.timeseries(universe.atoms)
+
+    universe2 = Universe(PDB_small, coordinates,
+                         format=ArrayReader)
```

Comment:
*I didn't have much time to read this yet*

Why the two step process? You could also add a kwarg to the Universe `__init__` to tell it directly that the coordinates should be loaded in memory.

```python
universe = mda.Universe(PSF, DCD, in_memory=True)
```